sklearn/model_selection/_search.py

```diff
@@ -788,9 +834,29 @@ def _store(key_name, array, weights=None, splits=False, rank=False):
         # Store a list of param dicts at the key 'params'
         results['params'] = candidate_params
 
-        # NOTE test_sample counts (weights) remain the same for all candidates
-        test_sample_counts = np.array(test_sample_counts[:n_splits],
-                                      dtype=np.int)
+        # training train_sample_weight_sums needs to be done first because
+        # test_sample_counts overrides itself in the IF statement below with
+        # test_sample_weight_sums.
+        if self.return_train_score:
+            if train_sample_weight_sums is None:
+                samples = int(np.sum(test_sample_counts[:n_splits]))
```

Comment:
These two lines are the hairiest of all.  I'm not 100% certain about whether this is the right way to do it but unfortunately, I don't see a better way because the total dataset size isn't maintained anywhere inside the function.  Any thoughts?

Reply:
Can we not store the total data set size in some way and pass it in?

Reply:
I'm not sure if this is possible.  The reason I say this is that we can't necessarily just take the data set size because I don't think that the cross validation iterator (the `cv` parameter in `GridSearchCV`) must iterate over the entire dataset.  So it should be based on the sizes defined by `cv`.

Reply:
Should I put a comment to this effect in the code?